packages/admin/resources/views/components/layouts/app/sidebar/item.blade.php

```diff
@@ -8,12 +9,18 @@
     'url',
 ])
 
-<li
-    @class([
-        'filament-sidebar-item overflow-hidden',
-        'filament-sidebar-item-active' => $active,
-    ])
->
+@php
+    $iconClasses = \Illuminate\Support\Arr::toCssClasses([
```

Comment:
No need to extract if only used in 1 place.

Reply:
There's no need to extract the classes into an `$iconClasses` variable if that variable is only used in a single place in the template. Or am I missing something?

Reply:
**iconColor** have 5 possible values [ primary,danger,secondary,success or primary ] , so without the code provided how i can select the color ?

in the case where i pass the color class directly like "text-primary-500" sure i dont need the code i provided above .

NavigationItem::make('Analytics 1')
                    **->iconColor('danger')** // **_i pass "danger" not text-danger-500 , so without this code above how i can select the correct color that match with the parameter i passed "danger"._**

Reply:
> **iconColor** have 5 possible values [ primary,danger,secondary,success or primary ] , so without the code provided how i can select the color ?
> 
> in the case where i pass the color class directly like "text-primary-500" sure i dont need the code i provided above .
> 
> NavigationItem::make('Analytics 1') **->iconColor('danger')** // **_i pass "danger" not text-danger-500 , so without this code above how i can select the correct color that match with the parameter i passed "danger"._**

and if i'm missing something please could u tell me some other way to implement the solution i nedded in the best way that exist

Reply:
The API doesn't need changing. I'm just saying storing it in the $iconClasses variable in the view isn't needed as the classes can be applied to the HTML element using the `@class()` Blade directive directly. 🙂

Reply:
it's done,thanks.